### ISSUE TITLE
[BUG] Asymetric inertia gradients

### ISSUE THREAD

**Issue by JonathanKuelz** [NVIDIA/warp#585]
### Bug Description

I am running a simulation where a sphere rolls down a tilted plane. After some time steps, I compute the distance traveled and compute gradients of this distance with respect to the original inertia matrix.

What I get is the following:
```
print(sim.model.body_inertia.grad)
[
[-1.3617388e-07  5.5382614e-05  7.1924973e-08]
[ 5.4380315e-12  6.5725419e-07  3.9953427e-12]
[-1.1512999e-07  2.5967754e-02 -5.2107964e-07]
]
```

First of all, the gradients are suspiciously small*. What really is suspicious though is the asymmetry in the inertia gradients. I double-checked, and the inertia matrix is symmetric, so I am wondering: How can the gradients be different for `Ixy` and `Iyx`?

I am working with the SemiImplicitIntegrator. The only reason I have for a) the gradient tensor being asymmetric while b) it not being zero on the lower left off-diagonal is that somewhere in the simulation step, there must be operations that treat the inertia as a matrix while other methods only work with the unique values of the inertia matrix.

The tape for the backpropagation looks reasonable:

![Image](https://github.com/user-attachments/assets/ff95c21a-7278-4b81-a216-4f2863cc7ea2)

*Some numerical values:
- Sphere radius: 0.4
- Sphere mass: 265
- Distance traveled: 4.36
- Ixx = Iyy = Izz = 1.69

### System Information

Warp version 1.6.0

**Comment by JonathanKuelz** [comment]
I think I figured out (most) the problem:

1. The symmetry of the inertia tensor is *not* explicitly taken into account. Even though I believe the gradient matrix _should_ be a symmetric, it is not. This can be explained by values like `I_xy` and `I_yx` being treated separately. I attached a small script to reproduce an asymmetric inertial gradient matrix with just the `integrate_bodies` function where problem (2) is already fixed below.
2. When calling `builder.finalize()`, warp computes the attributes `body_inv_inertia` and `body_inv_mass`. This happens (with some exceptions, but I think this holds for most use cases) *outside* of the tape, so gradients to not backpropagate from computations that use the inverse inertia to the original inertial values. Later on, these two matrices are both used - for example in `integrate_rigid_bodies`. The attribute `model.body_inertia.grad` thus does *not* capture gradients for any computation performed with `inv_inertia`.

Minimum reproducible demo (computing inv_inertia manually to avoid the second issue reported):
```python
import numpy as np
import warp as wp
import warp.sim

@wp.kernel
def invert(arr: wp.array(dtype=wp.mat33f), out: wp.array(dtype=wp.mat33f)):
    """Computes the inverse of a 3x3 matrix."""
    tid = wp.tid()
    out[tid] = wp.inverse(arr[tid])

@wp.kernel
def integrate_body(
    inertia: wp.array(dtype=wp.mat33),
    inv_inertia: wp.array(dtype=wp.mat33),
    loss: wp.array(dtype=wp.float32),
):
    tid = wp.tid()
    q = wp.transform(wp.vec3(0., 0., 0.), wp.quat_identity())
    qd = wp.spatial_vector(0., 0., 0., 0., 0., 0.)
    f = wp.spatial_vector(1., 2., 3., -1., -1.5, -2.)
    com = wp.vec3(1., 0., 0.)
    inv_mass = 1.
    gravity = wp.vec3(0., 0., -9.81)
    angular_damping = 0.1
    dt = 0.1
    q_new, qd_new = wp.sim.integrator.integrate_rigid_body(q, qd, f, com, inertia[tid], inv_mass, inv_inertia[tid],
                                                           gravity, angular_damping, dt)
    loss[0] = qd_new[0] + qd_new[1] + qd_new[2] + qd_new[3] + qd_new[4] + qd_new[5]

inertia = wp.array(np.array([[1.0, 0.1, 0.2], [0.1, 2.0, 0.3], [0.2, 0.3, 3.0]]), dtype=wp.mat33f)
inertia.requires_grad = True
inv_inertia = wp.zeros_like(inertia)
loss = wp.zeros(1, dtype=wp.float32, requires_grad=True)
with wp.Tape() as tape:
    wp.launch(invert, dim=1, inputs=[inertia], outputs=[inv_inertia])
    wp.launch(integrate_body, dim=1, inputs=[inertia, inv_inertia], outputs=[loss])
tape.backward(loss)
print(inertia.grad)
```

Shows
```
[[[-0.06710325 -0.0751805  -0.07828714]
  [-0.03089282 -0.03461141 -0.03604162]
  [-0.01696573 -0.0190079  -0.01979335]]]
```

**Comment by JonathanKuelz** [comment]
For the featherstone integrator, the method `self.allocate_model_aux_vars(model)` additionally needs to be included in the tape scope if gradients w.r.t. inertial properties are of interest.
